src/store/slices/userSlice/action.ts

```diff
@@ -0,0 +1,32 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { createUserWithEmailAndPassword, getAuth, signInWithEmailAndPassword } from 'firebase/auth';
+import { IUser } from 'models/IUser';
+
+export const loginOrCreateUser = createAsyncThunk(
+	'user/loginOrCreate',
+	async (user: { uEmail: string, uPassword: string, isNewUser: boolean }, thunkApi) => {
+		const auth = getAuth();
+		const { uEmail, uPassword, isNewUser } = user;
+		try {
+			const { user } = isNewUser
+				? await createUserWithEmailAndPassword(auth, uEmail, uPassword)
+				: await signInWithEmailAndPassword(auth, uEmail, uPassword);
+
+			// @ts-ignore
```

Comment:
why? I think you can add some checker here (google [TypeGuard](https://www.google.com/search?q=TypeGuard&sourceid=chrome&ie=UTF-8))